internal/common/shared/gifts.go

```diff
@@ -1,7 +1,28 @@
 package shared
 
-// GiftDict is a dictonary of gift
-type GiftDict = map[ClientID]uint
+// GiftRequest contains the details of a gift request from an island to another
+type GiftRequest struct {
+	RequestingTeam ClientID
+	OfferingTeam   ClientID
+	RequestAmount  Resources
+}
+
+// GiftRequestDict is a dictionary of offers that a client will make to other clients.
+// A layer of redundancy exists, in that the key and the OfferingTeam encode the same information,
+// but this may prove useful in implementation.
+type GiftRequestDict = map[ClientID]GiftRequest
+
+// GiftOffer contains the details of a gift offer from an island to another
+type GiftOffer struct {
+	ReceivingTeam ClientID
+	OfferingTeam  ClientID
+	OfferAmount   Resources
+}
```

Comment:
Could these 2 maybe be merge into just `type Gift struct`?

Reply:
Shit lol I didn't even realise how similar they were until now. I'll leave them as-is for now, we might need them to be separate when we make sure all edge cases are covered. (which is probably why I thought we needed 2 in the first place).